### PR TITLE
m_geoipban ipv6 support

### PR DESCRIPTION
**Pull request by H7-25**
Added ipv6 support to m_geoipban module

### CODE REVIEW

File: 2.0/m_geoipban.cpp [hunk 1]
```diff
@@ -97,12 +113,24 @@ class ModuleGeoIPBan : public Module
...
+        if (inet_pton(AF_INET, dst->GetIPString(), buf)) {
+
+    		d = GeoIP_country_name_by_addr(gi, dst->GetIPString());
+    		if (!d)
+    			d = "UNKNOWN";
+        } else if (inet_pton(AF_INET6, dst->GetIPString(), buf)) {
+ 
+     		d = GeoIP_country_name_by_addr_v6(fi, dst->GetIPString());
+    		if (!d)
+    			d = "UNKNOWN";
```
Comment (SISheogorath) [inline 2.0/m_geoipban.cpp]: Can be moved out of the outer if and reduce code duplication.

File: 2.0/m_geoipban.cpp [hunk 2]
```diff
@@ -97,12 +113,24 @@ class ModuleGeoIPBan : public Module
...
-		if (!d)
-			d = "UNKNOWN";
+        char buf[16];
+        const char* d = "UNKNOWN";
+        
+        if (inet_pton(AF_INET, dst->GetIPString(), buf)) {
+
+    		d = GeoIP_country_name_by_addr(gi, dst->GetIPString());
+    		if (!d)
+    			d = "UNKNOWN";
```
Comment (SISheogorath) [inline 2.0/m_geoipban.cpp]: Can be moved out of the outer if and reduce code duplication.

File: 2.0/m_geoipban.cpp [hunk 3]
```diff
@@ -40,12 +40,25 @@ class ModuleGeoIPBan : public Module
...
+        if (inet_pton(AF_INET, user->GetIPString(), buf)) {
+
+    		c = GeoIP_country_code_by_addr(gi, user->GetIPString());
+    		if (!c)
+    			c = "UNK";
+        } else if (inet_pton(AF_INET6, user->GetIPString(), buf)) {
+ 
+     		c = GeoIP_country_code_by_addr_v6(fi, user->GetIPString());
+    		if (!c)
+    			c = "UNK";
```
Comment (SISheogorath) [inline 2.0/m_geoipban.cpp]: Can be moved out of the outer if and reduce code duplication.

File: 2.0/m_geoipban.cpp [hunk 4]
```diff
@@ -40,12 +40,25 @@ class ModuleGeoIPBan : public Module
...
-		if (!c)
-			c = "UNK";
+        char buf[16];
+        const char* c = "UNK";
+        
+        if (inet_pton(AF_INET, user->GetIPString(), buf)) {
+
+    		c = GeoIP_country_code_by_addr(gi, user->GetIPString());
+    		if (!c)
+    			c = "UNK";
```
Comment (SISheogorath) [inline 2.0/m_geoipban.cpp]: Can be moved out of the outer if and reduce code duplication.

File: 2.0/m_geoipban.cpp [hunk 5]
```diff
@@ -40,12 +40,25 @@ class ModuleGeoIPBan : public Module
 {
 	LocalStringExt ext;
 	GeoIP* gi;
-
+    GeoIP* fi;
```
Comment (SISheogorath) [inline 2.0/m_geoipban.cpp]: Formatting doesn't match above

File: 2.0/m_geoipban.cpp [hunk 6]
```diff
@@ -40,10 +40,23 @@ class ModuleGeoIPBan : public Module
...
-
+	GeoIP* fi;
+	
 	std::string* SetExt(User* user)
 	{
-		const char* c = GeoIP_country_code_by_addr(gi, user->GetIPString());
+		char buf[16];
+		const char* c = "UNK";
+		
+		if (inet_pton(AF_INET, user->GetIPString(), buf)) {
```
Comment (SaberUK) [inline 2.0/m_geoipban.cpp]: You can check if `client_sa.sa.sa_family` is `AF_INET` or `AF_INET6` instead of doing this.

File: 2.0/m_geoipban.cpp [hunk 7]
```diff
@@ -62,6 +75,9 @@ class ModuleGeoIPBan : public Module
 		gi = GeoIP_new(GEOIP_STANDARD);
 		if (gi == NULL)
 			throw ModuleException("Unable to initialize geoip, are you missing GeoIP.dat?");
+		fi = GeoIP_open("/usr/share/GeoIP/GeoIPv6.dat", GEOIP_STANDARD);
```
Comment (SaberUK) [inline 2.0/m_geoipban.cpp]: Is there no way to detect this file location automatically like what GeoIP_new does?

Comment (SaberUK) [reply 2.0/m_geoipban.cpp]: Looking at the library it seems like you can do `GeoIP_open_type(GEOIP_COUNTRY_EDITION_V6, GEOIP_STANDARD);` maybe?

File: 2.0/m_geoipban.cpp (hunk 6 above)
Comment (H7-25) [reply 2.0/m_geoipban.cpp]: Are you sure user->client_sa.sa.sa_family is available in insp20 ?

Comment (SaberUK) [reply 2.0/m_geoipban.cpp]: Yes.

File: 2.0/m_geoipban.cpp [hunk 8]
```diff
@@ -40,10 +40,23 @@ class ModuleGeoIPBan : public Module
...
+	
 	std::string* SetExt(User* user)
 	{
-		const char* c = GeoIP_country_code_by_addr(gi, user->GetIPString());
+		const char* c = NULL;
+		
+		switch (user->client_sa.sa.sa_family)
+		{
+			case AF_INET:
+                c = GeoIP_country_code_by_addr(gi, user->GetIPString());
```
Comment (SaberUK) [inline 2.0/m_geoipban.cpp]: The indentation of this (and the line below) is broken (i.e. using spaces instead of tabs).

File: 2.0/m_geoipban.cpp [hunk 9]
```diff
@@ -97,12 +113,24 @@ class ModuleGeoIPBan : public Module
 		if (!cc)
 			cc = SetExt(dst);
 
-		const char* d = GeoIP_country_name_by_addr(gi, dst->GetIPString());
+		const char* d = NULL;
+		
+		switch (dst->client_sa.sa.sa_family)
+		{
+			case AF_INET:
+                d = GeoIP_country_name_by_addr(gi, dst->GetIPString());
```
Comment (SaberUK) [inline 2.0/m_geoipban.cpp]: Indentation is broken here + below.

File: 2.0/m_geoipban.cpp [hunk 10]
```diff
@@ -1,7 +1,7 @@
 /*
  * InspIRCd -- Internet Relay Chat Daemon
  *
- *   Copyright (C) 2013 Filippo Cortigiani <simos@simosnap.org>
+ *	 Copyright (C) 2013 Filippo Cortigiani <simos@simosnap.org>
```
Comment (SaberUK) [inline 2.0/m_geoipban.cpp]: The indentation here is broken.

File: 2.0/m_geoipban.cpp [hunk 11]
```diff
@@ -13,7 +13,7 @@
  * details.
  *
  * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * along with this program.	 If not, see <http://www.gnu.org/licenses/>.
```
Comment (SaberUK) [inline 2.0/m_geoipban.cpp]: This doesn't need to change?

Comment (H7-25) [reply 2.0/m_geoipban.cpp]: Ops ... something goes wrong using coda2 plugin converting identation ... this night i'll make a new commit.

File: 2.0/m_geoipban.cpp [hunk 12]
```diff
@@ -1,7 +1,7 @@
 /*
  * InspIRCd -- Internet Relay Chat Daemon
  *
- *   Copyright (C) 2013 Filippo Cortigiani <simos@simosnap.org>
+ * Copyright (C) 2013 Filippo Cortigiani <simos@simosnap.org>
```
Comment (SaberUK) [inline 2.0/m_geoipban.cpp]:
```suggestion
 *   Copyright (C) 2013 Filippo Cortigiani <simos@simosnap.org>
```

File: 2.0/m_geoipban.cpp [hunk 13]
```diff
@@ -13,7 +13,7 @@
  * details.
  *
  * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
```
Comment (SaberUK) [inline 2.0/m_geoipban.cpp]:
```suggestion
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

**Pull request merged**